src/content/blog/astro-5110.mdx

```diff
@@ -0,0 +1,108 @@
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## CSP with adapters
+
+Up until now, static pages could only serve the Content Security Policy (CSP) header via the `<meta http-equiv="content-security-poligy">` element. This was an inconvenience for some users, because some CSP directives such as `report-uri`, `frame-ancestors`, and `sandbox` are not supported in a `<meta>` element.
+
+From Astro 5.11.0, the official `@astrojs/node`, `@astro/netlify` and `@astro/vercel` adapters gained support for specifying custom headers for prerendered pages. This means that these adapters can now serve the CSP header for static pages using the `Response` headers, allowing support for all directives, even on prerendered pages.
+
+This is a new experimental feature available as a new adapter configuration option `experimentalStaticHeaders`:
+
+```js title="astro.config.mjs" ins={10}
+import { defineConfig } from "astro/config";
+import node from "@astrojs/node";
+
+export default defineConfig({
+    experimental: {
+        csp: true
+    },
+    adapter: node({
+        mode: "standalone",
+        experimentalStaticHeaders: true
+    })
+})
+```
+
+Each adapter will use the hosting platform capabilities to enable this feature:
+- Netlify will use the [Framework API](https://docs.netlify.com/frameworks-api/#headers)
+- Vercel will use the [project configuration](https://vercel.com/docs/project-configuration#headers)
+- Node.js will use an internal implementation
+
+We will soon ship support for Cloudflare too, which will use the [`_headers` file](https://developers.cloudflare.com/pages/configuration/headers/).
+
+Under the hood, the adapters leverage a new Adapter API feature called [`experimentalStaticHeaders`](https://docs.astro.build/en/reference/adapter-reference/#experimentalstaticheaders). If you're an adapter developer, you can now provide full CSP capabilities to your users.
+
+## Disable HTML streaming
+
+The `@astrojs/node` adapter can now be configured to disable [HTML streaming](https://docs.astro.build/en/guides/on-demand-rendering/#html-streaming) for on demand pages with the new option `experimentalDisableStreaming`.
+
+HTML streaming helps with performance and generally provides a better visitor experience. In most cases, disabling streaming is not recommended.
+
+However, when you need to disable HTML streaming (e.g. your host only supports non-streamed HTML caching at the CDN level), you can now opt out of the default behavior:
+
+```js title="astro.config.mjs" ins={7}
+import { defineConfig } from 'astro/config';
+import node from '@astrojs/node';
+
+export default defineConfig({
+  adapter: node({
+    mode: 'standalone',
+    experimentalDisableStreaming: true
+  })
+});
+```
+
+## Bug fixes
+
+As always, we've been working hard on fixing issues since the [5.11 release](/blog/astro-590). See [the changelog](https://github.com/withastro/astro/blob/main/packages/astro/CHANGELOG.md) for all the details.
```

Comment:
```suggestion
As always, we've been working hard on fixing issues since the [5.10 release](/blog/astro-5100). See [the changelog](https://github.com/withastro/astro/blob/main/packages/astro/CHANGELOG.md) for all the details.
```